core/storage/btree.rs

```diff
@@ -1861,15 +1849,7 @@ impl BTreeCursor {
             let contents = page.get().contents.as_ref().unwrap();
             let cell_count = contents.cell_count();
             if cell_count == 0 {
-                self.stack.set_cell_index(0);
-                match seek_op.iteration_direction() {
-                    IterationDirection::Forwards => {
-                        return self.next();
-                    }
-                    IterationDirection::Backwards => {
-                        return self.prev();
-                    }
-                }
+                return Ok(CursorResult::Ok(SeekResult::NotFound));
```

Comment:
honestly I don't know why this was this way, I also removed it in the draft PR #1988 because I didn't get why it existed.